src/makeCalculator.js

```diff
@@ -37,7 +37,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(number) {
+      return this.result + number;
+    },
+
+    subtract(number) {
+      return this.result - number;
+    },
+
+    multiply(number) {
+      return this.result * number;
+    },
+
+    divide(number) {
+      return this.result / number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(operation, number) {
+      this.result = operation.call(this, number);
```

Comment:
```suggestion
      operation.call(this, number);
```
now you can simply execute function, without setting it's return to result